core/src/main/java/de/danoeh/antennapod/core/util/playback/AudioPlayer.java

```diff
@@ -64,4 +65,17 @@ public void setAudioTrack(int track) {
     public int getSelectedAudioTrack() {
         return -1;
     }
+
+    @Override
+    public void setDataSource(String streamUrl, String username, String password) {
+        try {
+            setDataSource(streamUrl);
```

Comment:
That looks like it could get the media player into an inconsistent state where AntennaPod thinks it is playing but it actually crashed. I think it would be better to pass the exceptions to the caller.